src/units/unit.cpp

```diff
@@ -2442,14 +2442,20 @@ void unit::add_modification(const std::string& mod_type, const config& mod, bool
 {
 	bool generate_description = mod["generate_description"].to_bool(true);
 
+	config* target = nullptr;
+
 	if(no_add == false) {
-		modifications_.add_child(mod_type, mod);
+		target = &modifications_.add_child(mod_type, mod);
+		target->remove_children("effect");
 	}
 
-	bool set_poisoned = false; // Tracks if the poisoned state was set after the type or variation was changed.
 	config type_effect, variation_effect;
```

Comment:
These two configs seem to be unused now.

I haven't managed to understand the code yet, when are the variations now applied?

Reply:
in order, like other effects, as part of the loop here.